Tests/tests/test_ScreenCapture.cpp

```diff
@@ -0,0 +1,192 @@
+
+        PluginHost::IFactories::Assign(&factoriesImplementation);
+
+        dispatcher = static_cast<PluginHost::IDispatcher*>(
+            plugin->QueryInterface(PluginHost::IDispatcher::ID));
+        dispatcher->Activate(&service);
+    }
+    virtual ~ScreenCaptureEventTest() override
+    {
+        dispatcher->Deactivate();
+        dispatcher->Release();
+
+        PluginHost::IFactories::Assign(nullptr);
+
+        plugin->Deinitialize(nullptr);
+    }
+};
+
+class ScreenCaptureFrameBufferTest : public ScreenCaptureEventTest {
+protected:
+    FrameBufferApiImplMock frameBufferApiImplMock;
+
+    ScreenCaptureFrameBufferTest()
+        : ScreenCaptureEventTest()
+    {
+        FrameBufferApi::getInstance().impl = &frameBufferApiImplMock;
+    }
+    virtual ~ScreenCaptureFrameBufferTest() override
+    {
+        FrameBufferApi::getInstance().impl = nullptr;
+    }
+};
+
+TEST_F(ScreenCaptureTest, RegisteredMethods)
+{
+    EXPECT_EQ(Core::ERROR_NONE, handler.Exists(_T("uploadScreenCapture")));
+}
+
+TEST_F(ScreenCaptureFrameBufferTest, FrameBufferUpload)
+{
+    FBContext fbcontext;
+    PixelFormat pixelFormat = {32, 24, 0, 1, 255, 255, 255, 16, 8, 0};
+    vnc_uint8_t* frameBuffer = (vnc_uint8_t*) malloc(5120 * 720);
+    memset(frameBuffer, 0xff, 5120 * 720);
+    
+    Core::Event uploadComplete(false, true);
+    
+    EXPECT_CALL(frameBufferApiImplMock, fbCreate(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext** fbctx) {
+                *fbctx = &fbcontext;
+                return ErrNone;
+            }));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbInit(::testing::_, ::testing::_, ::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx, VncServerFramebufferAPI* server, void* serverctx) {
+                if (nullptr != server)
+                {
+                    server->framebufferUpdateReady(&fbcontext);
+                    server->framebufferDetailsChanged(&fbcontext, nullptr, 1280, 720, 5120, nullptr);
+                    server->paletteChanged(&fbcontext, nullptr);
+                    server->logMsg(&fbcontext, "msg");
+                }
+                return ErrNone;
+            }));
+
+    ON_CALL(frameBufferApiImplMock, fbGetPixelFormat(::testing::_))
+        .WillByDefault(
+            ::testing::Return(&pixelFormat));
+
+    ON_CALL(frameBufferApiImplMock, fbGetWidth(::testing::_))
+        .WillByDefault(
+            ::testing::Return(1280));
+
+    ON_CALL(frameBufferApiImplMock, fbGetHeight(::testing::_))
+        .WillByDefault(
+            ::testing::Return(720));
+
+    ON_CALL(frameBufferApiImplMock, fbGetStride(::testing::_))
+        .WillByDefault(
+            ::testing::Return(5120));
+
+    EXPECT_CALL(frameBufferApiImplMock, fbGetFramebuffer(::testing::_))
+        .Times(1)
+        .WillOnce(::testing::Invoke(
+            [&](FBContext* fbctx) {
+                return frameBuffer;
```

Comment:
please use `Return(frameBuffer)` and `using testing::Return;` where it simply  returns